### DISCUSSION TITLE
How to get the last entry of (named) time series

### DISCUSSION THREAD

**Discussion by Th3r3sa** [ravendb/ravendb#15761] [Q&A] [answered]
How can I get the last entry of a (named) time series? (I.e the entry with the largest timestamp).
As I don't know when the last entry was made I don't want to use "from date" and "to date".
I am using the C# client api.
The best solution that I see, is to save the timestamp of the last entry and use that to set "from date" and "to date" in my query. 
But perhaps there is a better way?
Kind regards
/Theresa

**Comment by ayende** [comment, marked as answer]
Something like:

```
    RavenQuery.TimeSeries(p, "HeartRate")
                                .FromLast(g => g.Minutes(30))
                                .GroupBy(g => g.Minutes(10))
                                .Select(x => new
                                {
                                    Min = x.Min(),
                                    Max = x.Max(),
                                    Avg = x.Average()
                                })
                                .ToList())
```

Here is what this looks like in terms of the query:

```
from 'Employees' 
where id() == 'employees/7-A'
select timeseries(from 'HeartRates' last '1 second')
```
